lib/redis/connection/ruby.rb

```diff
@@ -351,6 +352,13 @@ def get_tcp_keepalive
         end
       end
 
+      # disables Nagle's Algorithm, prevents multiple round trips with MULTI
+      def set_tcp_nodelay
+        if [:IPPROTO_TCP, :TCP_NODELAY].all?{|c| Socket.const_defined? c}
```

Comment:
Why do you check for these constants ? Are they missing on some implementations ?

If they really have to be checked for existence, I'd rather use `defined? Socket::TCP_NODELAY && defined? Socket::IPPROTO_TCP `.

Reply:
I did actually deliberate over using the syntax of the [conditional definition](https://github.com/redis/redis-rb/blob/1b527b1e536008aa19e11d64390e3d96c936a80e/lib/redis/connection/ruby.rb#L327)  of the similar method _set_tcp_keepalive_, but proceeded assuming it to be some form of precedent/convention. 

However, having checked the history I see that the code there is 7 years old and may just have been convenient at the time. I was actually wondering whether to follow up this pull request with another to remove that check (and perhaps perform the same conditional execution in my second commit?) around _set_tcp_keepalive_. However, it is an **optional** socket option with no default being set by Ruby Redis - but the conditional definition seems a bit peculiar.

I also wondered what the purpose is - as surely the constant definitions in Socket might not actually relate to the given subclass (i.e. trying to set TCP_NODELAY on a UNIXSocket actually generates an error)?

Anyway, happy to remove the seemingly redundant check.

Reply:
Ok, I think the code you point out is interesting. I'm not 100% sure why it checks, but at first sight I'd say it's for non POSIC platforms support. Or at least it's what I can infer from looking at MRI's source: https://github.com/ruby/ruby/blob/c5eb24349a4535948514fe765c3ddb0628d81004/ext/socket/option.c#L1288

In this case I believe it's best to do it the same way as the code you pointed, i.e. at definition time:

```ruby
  if [:IPPROTO_TCP, :TCP_NODELAY].all?{|c| Socket.const_defined? c}
      def set_tcp_nodelay        
        @sock.setsockopt(Socket::IPPROTO_TCP, Socket::TCP_NODELAY, 1)        
      end
  else
    def set_tcp_nodelay
    end
  end
```

Since we don't have a windows CI (yet?) to test it, I'd rather not get rid of those checks at this time.

Reply:
Also please squash your commits together.

Reply:
Thanks for drilling down into that, I was curious too.